synapse/handlers/federation_event.py

```diff
@@ -1014,6 +1146,8 @@ async def _process_received_pdu(
 
         await self._run_push_actions_and_persist_event(event, context, backfilled)
 
+        await self._handle_marker_event(origin, event)
```

Comment:
Moved this above the early-return when `backfilled`

TODO: link and quote the why